evalinstseg/util.py

```diff
@@ -100,11 +100,37 @@ def remove_empty_channels(labels):
 
     return np.array(tmp_labels)
 
+def expand_single_channel_to_stack(labels):
+    """
+    Expand a single-channel instance mask into a stack of binary masks.
+
+    Parameters
+    ----------
+    labels : np.ndarray
+        Array of shape (1, Z, Y, X) with integer instance IDs.
+
+    Returns
+    -------
+    np.ndarray
+        Array of shape (N, Z, Y, X), one binary mask per instance.
+    """
+    # Remove unnecessary channel dimension
+    vol = labels[0]
+
+    # Determine number of unique instances
+    max_id = np.max(vol)
+    if max_id <= 1:
+        return labels
+    
+    # Create binary masks for each instance
+    masks = [(vol == k) for k in range(1, max_id + 1)]
+    return np.stack(masks, axis=0).astype(labels.dtype)
+
 
 def check_fix_and_unify_ids(
         gt_labels, pred_labels, remove_small_components, foreground_only,
         dim_insts=[]):
-    """unify prediction and gt labelling styles
+    """unify prediction and gt labelling styles #! Wouldnt it make sense to also unify the representation of the UIDs (f.e, only channel per instamce instead of either single channel or channel per instance)
```

Comment:
Please create an issue for such comments/suggestions :)

First initial response without thinking about it too much would be that this could unnecessarily memory intensive for large 3d volumes.